_posts/2023-07-05-vanilla-os-orchid-devlog.md

```diff
@@ -0,0 +1,122 @@
+published: true
+date: 2023-07-05
+layout: article
+comments: true
+---
+
+A few weeks ago, we shared our progress in Orchid's development through our latest [Devlog](https://vanillaos.org/2023/06/11/vanilla-os-orchid-devlog.html). Among the many updates, we discussed [Apx v2](https://vanillaos.org/2023/06/11/vanilla-os-orchid-devlog.html#apx-v2-and-vso-v2) and how it was evolving to fit the developers needs better, while [VSO](https://vanillaos.org/2023/06/11/vanilla-os-orchid-devlog.html#vso-as-a-system-package-manager) would focus on meeting the needs of regular users. Crucial development of Apx v2 has finished and is available for testing now!
+
+## Apx v2
+
+As previously mentioned, Apx v2 sets itself apart from v1 with the  introduction of stacks. Stacks are sets of instructions that define how a subsystem gets created. While creating a stack, you can define the name, distribution, package manager, and list of packages to install during the subsystem's initialization phase.
+
+### Installation
+
+#### Method 1 (manual)
+
+To try out Apx v2, follow these instructions for a user-side installation:
+
+- Create a working directory (highly recommended to isolate Apx v2 from your system).
+
+- Obtain the latest Distrobox release (version [1.5.0.2](https://github.com/89luca89/distrobox/archive/refs/tags/1.5.0.2.tar.gz) or higher) and unpack it into a "distrobox" folder within your newly created work directory.
+
+- Download the latest Apx build from our [GitHub repository](https://github.com/Vanilla-OS/apx/releases/download/continuous/apx.tar.gz) and unpack the `apx` binary into `~/.local/bin`, renaming it to `apx2` to avoid collisions with previous versions.
+- Make the binary executable using the command `chmod +x apx2` from the correct directory.
+
+- Create the configuration file at `~/.config/apx/apx.json`:
+
+```json
+{
+  "apxPath": "/home/YOUR_USERNAME/.local/share/apx/",
+  "distroboxpath": "/home/YOUR_USERNAME/your/workdir/distrobox-folder/distrobox",
+  "storageDriver": "btrfs"
+}
+```
+
+Following these steps, you can start using Apx v2 by typing `apx2`.
+
+> Please note this is alpha software, and there may be bugs. We kindly request you to report any issues to our [GitHub repository](https://github.com/Vanilla-OS/apx/issues).
+
+Since this is a local installation of Apx, there won't be any initial contents. You can proceed manually by creating a package manager, a stack, and finally, a subsystem:
+
+```shell
+apx2 pkgmanagers new
+apx2 stacks new
+apx2 subsystems new
+apx2 <YOUR_NEW_SUBSYSTEM_NAME> --help
+```
+
+To simplify your initial experimentation with Apx v2, you can access some starter resources from our [vanilla-apx-configs repository](https://github.com/Vanilla-OS/vanilla-apx-configs/). Copy the contents into `~/.local/share/apx/` to populate Apx with initial resources like stacks and package managers. All major distributions are covered.
```

Comment:
Maybe move the steps on how to use apx after the two installation methods? People following the second installation method (which most people will, IMO) may not see this.

Reply:
> Maybe move the steps on how to use apx after the two installation methods? People following the second installation method (which most people will, IMO) may not see this.

Btw, this is done automatically in the last part of the script.

Reply:
> Btw, this is done automatically in the last part of the script.

What is done? The instructions on how to use apx are printed? Even then, I think it will be better to move the usage instructions after both the installation methods.

Reply:
> What is done? The instructions on how to use apx are printed? Even then, I think it will be better to move the usage instructions after both the installation methods.

How should I phrase it? 🤔

In the first method, you are suggested to create your own stacks and package managers, with a suggestion to optionally import them from the config repository. In the second method, the built-ins (from the config repo) are setup automatically so you don't need to setup them up again. Maybe, we can add this alone again `apx2 <YOUR_NEW_SUBSYSTEM_NAME> --help`